marklogic-data-hub/src/main/java/com/marklogic/hub/flow/impl/FlowRunnerImpl.java

```diff
@@ -592,4 +606,66 @@ public void setFlowManager(FlowManager flowManager) {
     public HubConfig getHubConfig() {
         return hubConfig;
     }
+
+    // Brings in artifact settings for running a step
+    private void applyArtifactSettings(Step step, Map<String, Object> stepConfig) {
```

Comment:
How about making this a protected method? Then you can easily unit test the various branches in this method without having to connect to ML or setup a test flow.